js/graticule.js

```diff
@@ -299,8 +299,13 @@ export function Graticule (paramMetricQHistoryReference, paramEle, ctx, offsetDi
     // console.log("x-axis labeling offset: " + (this.graticuleDimensions[1] + this.graticuleDimensions[3] + this.pixelsBottom /2));
     for (let i = 0; i < yAxisSteps.length; ++i) {
       if (yPositions[i] >= this.graticuleDimensions[1]) {
-        const textWidth = this.ctx.measureText(yAxisSteps[i][1]).width
-        this.ctx.fillText(yAxisSteps[i][1], this.graticuleDimensions[0] - textWidth - this.pixelsLeft, yPositions[i] + 4)
+        this.ctx.textAlign = 'right'
+        let value = yAxisSteps[i][1]
+        if (value >= 10000) {
+          value = value.toString().substr(0, 1) + '.' + value.toString().substr(1, 1) + ' × 10^' + Math.floor(Math.log(value) / Math.log(10))
```

Comment:
I think we might as well use [`log10`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/log10). (obsolete by comment below.)

Reply:
In general, prefer distinct const's (e.g. `numericValue` and `displayValue`). It makes the code easier to reason about.

Reply:
There's a lot of assumptions here about what comes out of `toString`, that might be true but are a bit fragile. Like `toString` should never use thousand's separators.
I suggest we actually use `.toExponential` instead of doing these calculations on our own. Yes, it's different with `e` vs `10^`. Now you could use that and split the string by "e" - but I really don't like the `10^x` (I prefer the superscript). But plotting a superscript with `fillText`... is scary.

So, just using `toExponential(2)` seems like a good compromise. (2 fraction digits seems slightly better to me).